pkg/network/networker.go

```diff
@@ -35,26 +39,102 @@ import (
 
 const (
 	// ZDBIface is the name of the interface used in the 0-db network namespace
-	ZDBIface = "zdb0"
+	ZDBIface     = "zdb0"
+	wgPortDir    = "wireguard_ports"
+	networkDir   = "networks"
+	ipamLeaseDir = "ndmz-lease"
+	ipamPath     = "/var/cache/modules/networkd/lease"
+)
+
+const (
+	mib = 1024 * 1024
 )
 
 type networker struct {
-	identity   pkg.IdentityManager
-	storageDir string
-	tnodb      TNoDB
-	portSet    *set.UintSet
+	identity     pkg.IdentityManager
+	networkDir   string
+	ipamLeaseDir string
+	tnodb        TNoDB
+	portSet      *set.UintSet
 }
 
 // NewNetworker create a new pkg.Networker that can be used over zbus
-func NewNetworker(identity pkg.IdentityManager, tnodb TNoDB, storageDir string) pkg.Networker {
+func NewNetworker(identity pkg.IdentityManager, tnodb TNoDB, storageDir string) (pkg.Networker, error) {
+
+	// create a directory on tmpfs to store the reserved port used for wireguard
+	wgDir, err := cache.VolatileDir(wgPortDir, 1*mib)
+	if err != nil && !os.IsExist(err) {
+		return nil, fmt.Errorf("failed to create wireguard_port cache directory: %w", err)
+	}
+
+	// create a directory on tmpfs to store the network deployed
+	nwDir, err := cache.VolatileDir(networkDir, 20*mib)
+	if err != nil && !os.IsExist(err) {
+		return nil, fmt.Errorf("failed to create networks cache directory: %w", err)
+	}
+
+	ipamLease, err := cache.VolatileDir(ipamLeaseDir, 10*mib)
```

Comment:
Can't we just create a single VolatileDir for the networker that is like 50M and that's it instead of 3 different mounts?

Reply:
indeed